app/uk/gov/hmrc/customs/notification/repo/NotificationWorkItemRepo.scala

```diff
@@ -210,12 +210,16 @@ class NotificationWorkItemMongoRepo @Inject()(mongo: MongoComponent,
     }
   }
 
-  override def permanentlyFailedByCsIdExists(csid: ClientSubscriptionId): Future[Boolean] = {
-    val selector = csIdAndStatusSelector(csid, PermanentlyFailed)
+  override def permanentlyFailedAndHttp5xxByCsIdExists(csid: ClientSubscriptionId): Future[Boolean] = {
+    val serverErrorCodeMin = 500
+    val selector = and(
+      gte(NotificationWorkItemFields.mostRecentPushPullHttpStatusFieldName, serverErrorCodeMin),
```

Comment:
csIdAndStatusSelector as first expression to decrease total results faster.

Reply:
Order doesn't seem matter when using $and, so I've left this one be.